rockcraft/oci.py

```diff
@@ -74,14 +74,19 @@ def from_docker_registry(
 
         The image is fetched from the registry at ``REGISTRY_URL``.
 
-        :param image_name: The image to retrieve, in ``name:tag`` format.
+        :param image_name: The image to retrieve, in ``name@tag`` format.
```

Comment:
I was a bit conflicted here one whether to accept `name:tag` or `name@tag`, but ended up with the latter for two reasons:
1) client code (the image service) would need to convert the `project.base` to `name:tag` every time, which feels like busywork that the class can do itself, and
2) it keeps things consistent because the `arch` parameter uses Debian architectures. So, the class accepts debian/craft parameters and converts them internally to docker/goarch.